client/modules/user/actions/userSingleActions.js

```diff
@@ -111,15 +129,176 @@ export function sendLogout() {
         'Accept': 'application/json'
         , 'Content-Type': 'application/json'
       }
+      , credentials: 'same-origin'
       , body: null
     })
+    // .then((res) => {
+    //   console.log("RECEIVED LOGOUT RESPONSE");
+    //   if(res.status == 200) {
+    //     console.log("success");
+    //     dispatch(receiveLogout({
+    //       success: true
+    //     }))
+    //   } else {
+    //     console.log("not success");
+    //     dispatch(receiveLogout({
+    //       success: false,
+    //       message: "Failed to log out."
+    //     }))
+    //   }
+    //   console.log("HERE 1");
+    //   console.log(res);
+    //   // return res;
+    // })
     .then(res => res.json())
+    .then(json => {
+      console.log("HERE 2");
+      console.log(json);
+      return json;
+    })
     .then(json => dispatch(receiveLogout(json)))
     .then((json) => {
       //if they hit this route, where should they redirect to?
       if(json.success) {
+        // alert("success");
+        browserHistory.push('/')
+      } else {
+        alert("An error occured while trying to log out. Please try again.");
+      }
+    })
+  }
+}
+
+export const REQUEST_FORGOT_PASSWORD = "REQUEST_FORGOT_PASSWORD"
+function requestForgotPassword(username) {
+  return {
+    type: REQUEST_FORGOT_PASSWORD
+    , username: username
+  }
+}
+
+export const RECEIVE_FORGOT_PASSWORD = "RECEIVE_FORGOT_PASSWORD"
+function receiveForgotPassword(json) {
+  return {
+    type: RECEIVE_FORGOT_PASSWORD
+    , success: json.success
+    , error: json.message
+    , receivedAt: Date.now()
+  }
+}
+
+export function sendForgotPassword(username) {
+  return dispatch => {
+    dispatch(requestForgotPassword(username))
+    return fetch('/api/users/requestpasswordreset', {
+      method: 'POST'
+      , headers: {
+        'Accept': 'application/json'
+        , 'Content-Type': 'application/json'
+      }
+      , credentials: 'same-origin'
+      , body: JSON.stringify({
+        email: username
+      })
+    })
+    .then(res => res.json())
+    .then(json => dispatch(receiveForgotPassword(json)))
+    .then((json) => {
+      //pop up an alert and then redirect them
+      // console.log("PASSWORD RESET REQUEST");
+      // console.log(json);
+      if(json.success) {
+        alert("You should receive an email shortly with password reset instructions.");
+        browserHistory.push('/')
+      } else {
+        alert("There was a problem reseting your password on the server. Please contact a site admin.");
+      }
+    })
```

Comment:
Something i've run into a lot with QikHome is handling callback messaging.  These alerts are UI and should be i the View compents.  If every action in Yote has this line: 
``` 
   .then((json) => {
      return json;
    })
```